ipa-core/src/protocol/basics/check_zero.rs

```diff
@@ -49,9 +52,9 @@ where
     let rv_share =
         semi_honest_multiply(ctx.narrow(&Step::MultiplyWithR), record_id, &r_sharing, v).await?;
     let rv = F::from_array(
-        &rv_share
-            .reveal(ctx.narrow(&Step::RevealR), record_id)
-            .await?,
+        &malicious_reveal(ctx.narrow(&Step::RevealR), record_id, None, &rv_share)
```

Comment:
it is maybe worth renaming this protocol to `check_zero_malicious` because it is no longer "default" implementation (can be made cheaper by using semi-honest reveal). We only use it in MAC-based validation, so it should be just a rename imo